FirebaseInAppMessaging/Sources/Flows/FIRIAMClientInfoFetcher.m

```diff
@@ -16,52 +16,44 @@
-  // concern in the fetch flow.
-  [iid tokenWithAuthorizedEntity:projectNumber
-                           scope:@"fiam"
-                         options:nil
-                         handler:^(NSString *_Nullable token, NSError *_Nullable error) {
-                           if (error) {
-                             FIRLogWarning(kFIRLoggerInAppMessaging, @"I-IAM190001",
-                                           @"Error in fetching iid token: %@",
-                                           error.localizedDescription);
-                             completion(nil, nil, error);
-                           } else {
-                             FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM190002",
-                                         @"Successfully generated iid token");
-                             // now we can go ahead to fetch the id
-                             [iid getIDWithHandler:^(NSString *_Nullable identity,
-                                                     NSError *_Nullable error) {
-                               if (error) {
-                                 FIRLogWarning(kFIRLoggerInAppMessaging, @"I-IAM190004",
-                                               @"Error in fetching iid value: %@",
-                                               error.localizedDescription);
-                               } else {
-                                 FIRLogDebug(
-                                     kFIRLoggerInAppMessaging, @"I-IAM190005",
-                                     @"Successfully in fetching both iid value as %@ and iid token"
-                                      " as %@",
-                                     identity, token);
-                                 completion(identity, token, nil);
-                               }
-                             }];
-                           }
-                         }];
+- (void)fetchFirebaseInstallationDataWithProjectNumber:(NSString *)projectNumber
+                                        withCompletion:
+                                            (void (^)(NSString *_Nullable FID,
+                                                      NSString *_Nullable FISToken,
+                                                      NSError *_Nullable error))completion {
+  FIRInstallations *installations = [FIRInAppMessaging inAppMessaging].installations;
+
+  [installations authTokenWithCompletion:^(FIRInstallationsAuthTokenResult *_Nullable tokenResult,
+                                           NSError *_Nullable error) {
+    if (error) {
+      FIRLogWarning(kFIRLoggerInAppMessaging, @"I-IAM190006", @"Error in fetching FIS token: %@",
+                    error.localizedDescription);
+      completion(nil, nil, error);
+    } else {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM190007", @"Successfully generated FIS token");
+
+      [installations
+          installationIDWithCompletion:^(NSString *_Nullable identifier, NSError *_Nullable error) {
```

Comment:
Once backend is updated to accept only `FISToken` for authorization this could be simplified.